precompiles/proxy/Proxy.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.0;
+
+/// @author The Moonbeam Team
+/// @title The interface through which solidity contracts will interact with the Proxy pallet
+/// @custom:address 0x000000000000000000000000000000000000080b
+interface Proxy {
```

Comment:
Should we change to something like ProxyManager ?
The reason is because `Proxy` is often used in Smart Contract to designate the interface/contract proxying calls to other contract/libraries.

Reply:
On the other hand, I doubt this precompile will be often used by smart contract, it is more intended for direct use. so we can keep Proxy to be consistent